lldb/source/ValueObject/ValueObjectRecognizedValue.cpp

```diff
@@ -0,0 +1,310 @@
+//===-- ValueObjectRecognizedValue.cpp ------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "lldb/ValueObject/ValueObjectRecognizedValue.h"
+
+#include "lldb/Core/Address.h"
+#include "lldb/Core/Value.h"
+#include "lldb/DataFormatters/TypeRecognizer.h"
+#include "lldb/Symbol/CompilerType.h"
+#include "lldb/Symbol/Type.h"
+#include "lldb/Target/ExecutionContext.h"
+#include "lldb/Utility/DataExtractor.h"
+#include "lldb/Utility/LLDBLog.h"
+#include "lldb/Utility/Log.h"
+#include "lldb/Utility/Scalar.h"
+#include "lldb/Utility/Status.h"
+#include "lldb/ValueObject/ValueObject.h"
+#include "lldb/lldb-defines.h"
+#include "lldb/lldb-types.h"
+
+#include "llvm/Support/Error.h"
+
+#include <cstring>
+
+namespace lldb_private {
+class Declaration;
+} // namespace lldb_private
+
+using namespace lldb_private;
+
+ValueObjectRecognizedValue::ValueObjectRecognizedValue(
+    ValueObject &parent, lldb::DynamicValueType use_dynamic)
+    : ValueObject(parent), m_address(), m_dynamic_type_info(),
+      m_use_dynamic(use_dynamic) {
+  SetName(parent.GetName());
+}
+
+CompilerType ValueObjectRecognizedValue::GetCompilerTypeImpl() {
+  const bool success = UpdateValueIfNeeded(false);
```

Comment:
```suggestion
  const bool success = UpdateValueIfNeeded(/*update_format=*/false);
```
Also, I'm not sure that `success` is a good name. Inlining the call to `UpdateValueIfNeeded` into condition sounds better than coming up with a name